testing/testing-utils/src/generators.rs

```diff
@@ -131,36 +148,29 @@ pub fn generate_keys<R: Rng>(rng: &mut R, count: usize) -> Vec<Keypair> {
 /// transactions in the block.
 ///
 /// The ommer headers are not assumed to be valid.
-pub fn random_block<R: Rng>(
-    rng: &mut R,
-    number: u64,
-    parent: Option<B256>,
-    tx_count: Option<u8>,
-    ommers_count: Option<u8>,
-    requests_count: Option<u8>,
-    withdrawals_count: Option<u8>,
-) -> SealedBlock {
+pub fn random_block<R: Rng>(rng: &mut R, block_params: BlockParams) -> SealedBlock {
```

Comment:
I think we can leave the `number` argument as-is, because it's required. And have the rest collapsed into a new struct with optional fields
```suggestion
pub fn random_block<R: Rng>(rng: &mut R, number: u64, block_params: BlockParams) -> SealedBlock {
```